pyat/at/lattice/particle_object.py

```diff
@@ -75,3 +92,59 @@ def rest_energy(self) -> numpy.ndarray:
     def charge(self) -> float:
         """Particle charge [elementary charge]"""
         return self._charge
+
+    @property
+    def harmonic_number(self):
+        return self._harmn
+
+    @harmonic_number.setter
+    def harmonic_number(self, harmn):
+        self._harmn = harmn
+        self.fillpattern = 1
+
+    @property
+    def fillpattern(self):
+        return self._fillpattern
+
+    @fillpattern.setter
+    def fillpattern(self, bunches):
+        if (self._harmn is None):
+            warn(UserWarning("Harmonic number not set in Beam(), "
+                             "nbunch and fillpattern set to 1"))
+            fp = numpy.ones(1)
+        elif numpy.isscalar(bunches):
+            if bunches == 1:
+                fp = numpy.ones(1)
+            else:
+                bs = int(self._harmn/bunches)
+                fp = numpy.zeros((self._harmn,))
+                fp[0::bs] = 1
+        else:
+            assert len(bunches) == self._harmn, \
+                'Fill pattern has to be of shape ({0},)'.format(self._harmn)
+            assert numpy.all((bunches == 0) | (bunches == 1)), \
+                'Fill pattern can only contain 0 or 1'
+            fp = numpy.array(bunches)
+        if len(fp) != len(self._fillpattern):
+            self._weights = numpy.ones(len(self._fillpattern))
+        self._fillpattern = fp
+
+    @property
+    def weights(self):
+        return self._weights
+
+    @weights.setter
+    def weights(self, weights):
+        assert len(weights) == len(self._fillpattern), \
+            'Weights and fill pattern must have the same length'
+        self._weights = numpy.array(weights)
+
+    @property
+    def nbunch(self):
```

Comment:
this is a float but should be an int

Reply:
I don't think this really matters right?